### ISSUE TITLE
How to use subject access review in kubernetes-client?

### ISSUE THREAD

**Issue by raiRaiyan** [fabric8io/kubernetes-client#1455]
I want to use the kubernetes client to figure out if the client has the required permission to the resource. I can do this in kubectl using the following command:  

`kubectl auth can-i create deployments --namespace dev`

This api uses `SelfSubjectAccessReview` according to the [documentation](https://kubernetes.io/docs/reference/access-authn-authz/authorization/). Does the kubernetes-client support this? If yes, how do I use it?

Going through the source code, I was able to create the API object from `io.fabric8.kubernetes.api.model.authorization` like this:
```
LocalSubjectAccessReview review = new LocalSubjectAccessReviewBuilder()
                        .withNewSpec()
                        .withNewResourceAttributes()
                        .withResource("namespace")
                        .withVerb("create")
                        .endResourceAttributes()
                        .endSpec().build();
```

But i cannot use this either as there is no `Handler` for this type of a resource. Is there any way that I can use this API object for the time being?

**Comment by rohanKanojia** [comment]
Does it work when you use resource DSL for creation? Something like this:
```
KubernetesClient client = new DefaultKubernetesClient();
HasMetadata response = client.resource(review)..inNamespace("dev").createOrReplace();
```

**Comment by raiRaiyan** [comment]
No, it does not work. I get this error:
```
No handler found for object:LocalSubjectAccessReview(apiVersion=authorization.k8s.io/v1, kind=LocalSubjectAccessReview, metadata=null, spec=SubjectAccessReviewSpec(extra={}, groups=[], nonResourceAttributes=null, resourceAttributes=ResourceAttributes(group=null, name=null, namespace=null, resource=namespace, subresource=null, verb=create, version=null, additionalProperties={}), uid=null, user=null, additionalProperties={}), status=null, additionalProperties={})
```

I tried converting the review object to json and using load:
```
String json = mapper.writeValueAsString(review);
List<HasMetadata> result = ((DefaultKubernetesClient) client).load(new ByteArrayInputStream(json.getBytes())).fromServer().get();
```

I get the same error. It looks for a handler for this object type and fails. Following is the stack trace:
```
java.lang.NullPointerException: null
	at io.fabric8.kubernetes.client.dsl.internal.NamespaceVisitFromServerGetWatchDeleteRecreateWaitApplicableListImpl.acceptVisitors(NamespaceVisitFromServerGetWatchDeleteRecreateWaitApplicableListImpl.java:289)
	at io.fabric8.kubernetes.client.dsl.internal.NamespaceVisitFromServerGetWatchDeleteRecreateWaitApplicableListImpl.get(NamespaceVisitFromServerGetWatchDeleteRecreateWaitApplicableListImpl.java:267)
	at io.fabric8.kubernetes.client.dsl.internal.NamespaceVisitFromServerGetWatchDeleteRecreateWaitApplicableListImpl.get(NamespaceVisitFromServerGetWatchDeleteRecreateWaitApplicableListImpl.java:64)
	at com.app.kube.util.KubernetesUtils.verifyUserAccount(KubernetesUtils.java:59)
```

**Comment by rohanKanojia** [comment]
oh, Try to do it like this:
https://github.com/fabric8io/kubernetes-client/blob/1724b946300d824c6378c0f596bed1f826164ba5/kubernetes-tests/src/test/java/io/fabric8/openshift/client/server/mock/SubjectAccessReviewTest.java#L72

**Comment by raiRaiyan** [comment]
That works only for openshift client. I am using the kubernetes client, and it is not adaptable.

**Comment by rohanKanojia** [comment]
@raiRaiyan : Could you please create a PR to add this in kubenetes client dsl?

**Comment by raiRaiyan** [comment]
Ok. I will submit a PR once I get it working

**Comment by rohanKanojia** [comment]
Closed via #1466
